### PR TITLE
feat(amplify-category-auth): adding UserMigration trigger

### PR DESCRIPTION
**Pull request by pointnet**
*Issue #, if available:*

*Description of changes:*
Adds UserMigration trigger.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/amplify-category-auth/provider-utils/awscloudformation/triggers/UserMigration/function-template-dir/trigger-index.js [hunk 1]
```diff
@@ -0,0 +1,13 @@
+/*
+  this file will loop through all js modules which are uploaded to the lambda resource,
+  provided that the file names (without extension) are included in the "MODULES" env variable.
+  "MODULES" is a comma-delimmited string.
+*/
+
+exports.handler = (event, context, callback) => {
+  const modules = process.env.MODULES.split(',');
+  for (let i = 0; i < modules.length; i += 1) {
+    const { handler } = require(modules[i]);
```
Comment (haverchuck) [inline packages/amplify-category-auth/provider-utils/awscloudformation/triggers/UserMigration/function-template-dir/trigger-index.js]: To support node v10.x.x, I think we'll need to change this to:
`    const { handler } = require(`./${modules[i]}`);`

**Comment by kaustavghosh06** [comment]
@pointnet Do you have some time to review @haverchuck's requested changes?

**Comment by mdoesburg** [comment]
I need this feature. I've currently manually added a UserMigration trigger to my cloud formation, but this causes an error whenever I try to "Create or update Cognito user pool groups" through the CLI.

Someone please review this!

**Comment by mdoesburg** [comment]
@haverchuck Can this be reviewed again please?

**Comment by hisham** [comment]
+1. I need this feature as well...

**Comment by mdoesburg** [comment]
@mauerbac Is there a way we can get this added? Pretty sure this is the only Lambda Trigger that still can't be generated from the CLI, but I have no hopes for this PR ever being merged. 😂

**Comment by mauerbac** [comment]
@mdoesburg - thanks for the ping. Ive asked @brene to prioritize :)

**Comment by mdoesburg** [comment]
> @mdoesburg - thanks for the ping. Ive asked @brene to prioritize :)

Thank you!

**Comment by brene** [comment]
@mdoesburg - thanks for this awesome contrib! @UnleashedMind is actively looking into this PR. He has a few things on his plate right now but will look into it right after. Main areas that we'll look into is validation & testing.

File: packages/amplify-category-auth/provider-utils/awscloudformation/triggers/UserMigration/cloudformation-templates/UserMigration.json.ejs [hunk 2]
```diff
@@ -0,0 +1,193 @@
...
+                    "MODULES": {
+                        "Ref": "modules"
+                    },
+                    "REGION": { 
+                        "Ref": "AWS::Region"
+                    }
+                }
+            },
+            "Role": { "Fn::GetAtt" : ["LambdaExecutionRole", "Arn"] },
+            "Runtime": "nodejs8.10",
```
Comment (UnleashedMind) [inline packages/amplify-category-auth/provider-utils/awscloudformation/triggers/UserMigration/cloudformation-templates/UserMigration.json.ejs:100]: nodejs8.10 runtime is not longer supported by AWS Lambda.  Recomended version is:  nodejs12.x

**Comment by UnleashedMind** [comment]
Closing the PR because the PR fork is removed, change requests are not addressed and we can not update the PR either.

**Pull request closed without merging**